onnxruntime/hosting/main.cc

```diff
@@ -29,55 +27,26 @@ void test_request(const std::string& name, const std::string& version,
 }
 
 int main(int argc, char* argv[]) {
-  // TODO: create configuration class for all config related params
-  std::string model_path;
-  std::string address;
-  int port;
-  int threads;
-
-  po::options_description desc("Allowed options");
-  desc.add_options()("help,h", "Print a help message")("address,a", po::value(&address), "The base HTTP address")("port,p", po::value(&port), "HTTP port to listen to requests")("threads,t", po::value(&threads), "Number of http threads")("model_path,m", po::value(&model_path), "Path of the model file");
-
-  po::variables_map vm;
-  try {
-    po::store(po::parse_command_line(argc, argv, desc), vm);  // can throw
-
-    if (vm.count("help")) {
-      std::cout << "ONNX Hosting: host an ONNX model for inferencing with ONNXRuntime\n"
-                << std::endl
-                << desc << std::endl;
-      return EXIT_SUCCESS;
-    }
-
-    po::notify(vm);  // throws on error, so do after help
-  } catch (po::error& e) {
-    std::cerr << "An error with program arguments occurred with error: " << e.what() << std::endl
-              << std::endl;
-    std::cerr << desc << std::endl;
-    return EXIT_FAILURE;
-  } catch (const std::exception& e) {
-    std::cerr << "An unknown problem occurred with error: " << e.what() << std::endl
-              << std::endl;
-    return EXIT_FAILURE;
-  }
+  onnxruntime::hosting::ServerConfiguration config{};
```

Comment:
It is ok to have env and config separately in this PR. But in longer term, we may want to have the config part of the env. Then we could pass one "env" to http server for setup and inferencing.

Another benefit would be, you could use the logging in the `env` other than using `cout`/`cerr` in this class.

Reply:
Agreed. Later PR.